### ISSUE TITLE
Reflow dartdoc comments from LSP

### ISSUE THREAD

**Issue by dnfield** [dart-lang/sdk#35729]
Originally opened as https://github.com/Dart-Code/Dart-Code/issues/1413

----

One headache I have is that when I'm editing doc comments, I have to manually reflow them.  It'd be really nice if this plugin could help with that - so if I hit alt(opt) + q (or whatever hot key), it would automatically reflow the text in that doc comment block, e.g.:

```dart
/// Lorem ipsum dolor sit amet, ut mei impetus voluptaria reprimique. Timeam fastidii eos ei, malorum philosophia mei an. Ut oratio intellegam eos. Sale atomorum nec eu, detracto nominati voluptatibus nam ad. Ea falli ubique cetero per, ex sea pericula petentium repudiare. Partem rationibus interesset ei ius, ne argumentum definitiones mei.
```

Would become

```dart
/// Lorem ipsum dolor sit amet, ut mei impetus voluptaria reprimique. Timeam
/// fastidii eos ei, malorum philosophia mei an. Ut oratio intellegam eos. Sale
/// atomorum nec eu, detracto nominati voluptatibus nam ad. Ea falli ubique cetero
/// per, ex sea pericula petentium repudiare. Partem rationibus interesset ei ius,
/// ne argumentum definitiones mei.
```

And if I added a bunch of words to one of those lines later, I could again easily reflow the paragraph.

There's an implementation for VSCode that does **almost** what I want it to do in the [Reflow Markdown plugin](https://marketplace.visualstudio.com/items?itemName=marvhen.reflow-markdown).  Obviously that's in typescript and has no awareness of how to handle triple-slashes at the front of each line, but I imagine it wouldn't be super hard to implement that (which is why I originally opened this as an issue for Dart-Code).  At the same time, if this was implemented at the language server level it would benefit more users/IDEs.

/cc @DanTup @bwilkerson

**Comment by kevmoo** [comment]
@devoncarew @stereotype441 – thoughts on how you'd like to triage this? We don't have a label for LSP – should we?

**Comment by bwilkerson** [comment]
No, the LSP support is part of analyzer, so it should go in 'area-analyzer'. When I was triaging issues, I found it helpful to not have other labels added so that I could easily find the issues I hadn't yet looked at. I think that's what Bob's issue management doc also proposed. I don't, of course, know whether Paul works the same way.

**Comment by DanTup** [comment]
I think the title may also be slightly misleading; the makes sense for the server using both the existing and LSP protocols.

**Comment by srawlins** [comment]
I think that Dart's markdown package would be perfect for this. If the text under the cursor is a paragraph block, then it could reflow and be aware of inline syntax like code, etc.
